pkg/vm/vm.go

```diff
@@ -1270,27 +1268,15 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 		}
 
 	case opcode.RET:
-		oldCtx := v.istack.Pop().Value().(*Context)
-		rvcount := oldCtx.rvcount
-		oldEstack := v.estack
+		v.istack.Pop()
 
-		if rvcount > 0 && oldEstack.Len() < rvcount {
-			panic("missing some return elements")
-		}
 		if v.istack.Len() == 0 {
 			v.state = haltState
 			break
 		}
 
 		newEstack := v.Context().estack
-		if oldEstack != newEstack {
-			if rvcount < 0 {
-				rvcount = oldEstack.Len()
-			}
-			for i := rvcount; i > 0; i-- {
-				elem := oldEstack.RemoveAt(i - 1)
-				newEstack.Push(elem)
-			}
+		if v.estack != newEstack {
```

Comment:
Shouldn't we copy items from old stack to new stack here?

Reply:
Yes, we should, fixed. But also #1220 .